tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "compilerOptions": {
+    "outDir": "lib",
+    "allowJs": true,
+    "target": "es6",
+    "module": "CommonJS",
+    "esModuleInterop": true,
+  },
```

Comment:
We should add `"rootDir": "src", "sourceMap": false and "strict": true` within `compilerOptions`

Reply:
Yeah, as I mentioned above, I'm going to tighten up TS config to be strict after a first pass.